File: ParkByoengHun/1월 2주차/줄 서는 방법.js
function solution(n, k) {
    let answer = [];
    let array = Array.from({length: n}, (v, i) => i + 1);
    let nFacto = factorial(n);

    while (answer.length < n) {
        nFacto = nFacto / array.length;
        answer.push(...array.splice(Math.floor((k - 1) / nFacto), 1));
        k = k % nFacto;
    }

    return answer;
}

const factorial = (n) => {
    let answer = 1;
    for (let i = 1; i <= n; i ++) {
        answer *= i;
    }
    return answer;
}

console.log(solution(3, 5))

File: LeeNaYoung/1월 2주차/줄서는방법.py
def solution(n, k):
    answer = []
    line = [i for i in range(1,n+1)]
    k-=1
    # def fac(num):
    #     result = 1
    #     for i in range(1, num+1):
    #         result *= i
    #     return result
    import math
    for i in range(n-1,0,-1):
        m = math.factorial(i) #fac(i)
        answer.append(line[k//m])
        del line[k//m]
        k = k%m
    answer.append(line[0])
    return answer